File: ruby/aa/digital_root.rb
def digital_root(num)
  while num > 9
    num = digital_root_step(num)
  end

  num
end

def digital_root_step(num)
  root = 0
  while num > 0
    root += num % 10
    num /= 10
  end
  root
end

p digital_root(345)
p digital_root(13456)

File: js/aa/digitalRoot.js
function digitalRoot(num) {
  let dRoot = num;
  while (dRoot > 9) {
    dRoot = sumNum(dRoot);
  }
  return dRoot;
}

function sumNum(num) {
  let sum = 0;
  while (num > 0) {
    sum += num % 10;
    num = Math.floor(num / 10);
  }
  return sum;
}

console.log(digitalRoot(12345));
console.log(digitalRoot(345));
